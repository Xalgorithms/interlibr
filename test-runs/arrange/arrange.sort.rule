META
  VERSION "0.0.1"
  MAINTAINER "Don Kelly <karfai@gmail.com>";

REQUIRE refine:table0:0.0.1 AS table0;

ARRANGE table:table0 AS table_sort_a USING sort(a, "numeric");
ARRANGE table:table0 AS table_sort_a_dsc USING sort(a, "numeric", "descending");
ARRANGE table:table0 AS table_sort_b USING sort(b);
ARRANGE table:table0 AS table_sort_b_dsc USING sort(b, "alpha", "descending");
ARRANGE table:table0 AS table_sort_c USING sort(c);
ARRANGE table:table0 AS table_sort_bc USING sort(add(b, c), "alpha");
