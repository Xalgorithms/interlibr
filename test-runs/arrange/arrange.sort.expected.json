{
  "tables" : {
    "table" : {
      "table_sort_a" : [
        { "a" : 1, "b" : "C", "c": "A" },
        { "a" : 2, "b" : "A", "c": "P" },
        { "a" : 3, "b" : "E", "c": "B" },
        { "a" : 4, "b" : "A", "c": "Q" },
        { "a" : 5, "b" : "F", "c": "A" }
      ],
      "table_sort_a_dsc" : [
        { "a" : 5, "b" : "F", "c": "A" },
        { "a" : 4, "b" : "A", "c": "Q" },
        { "a" : 3, "b" : "E", "c": "B" },
        { "a" : 2, "b" : "A", "c": "P" },
        { "a" : 1, "b" : "C", "c": "A" }
      ],
      "table_sort_b" : [
        { "a" : 4, "b" : "A", "c": "Q" },
        { "a" : 2, "b" : "A", "c": "P" },
        { "a" : 1, "b" : "C", "c": "A" },
        { "a" : 3, "b" : "E", "c": "B" },
        { "a" : 5, "b" : "F", "c": "A" }
      ],
      "table_sort_b_dsc" : [
        { "a" : 5, "b" : "F", "c": "A" },
        { "a" : 3, "b" : "E", "c": "B" },
        { "a" : 1, "b" : "C", "c": "A" },
        { "a" : 4, "b" : "A", "c": "Q" },
        { "a" : 2, "b" : "A", "c": "P" }
      ],
      "table_sort_c" : [
        { "a" : 1, "b" : "C", "c": "A" },
        { "a" : 5, "b" : "F", "c": "A" },
        { "a" : 3, "b" : "E", "c": "B" },
        { "a" : 2, "b" : "A", "c": "P" },
        { "a" : 4, "b" : "A", "c": "Q" }
      ],
      "table_sort_bc" : [
        { "a" : 2, "b" : "A", "c": "P" },
        { "a" : 4, "b" : "A", "c": "Q" },
        { "a" : 1, "b" : "C", "c": "A" },
        { "a" : 3, "b" : "E", "c": "B" },
        { "a" : 5, "b" : "F", "c": "A" }
      ]
    }
  }
}
